fix(segment): nested interactives are not rendered

liamdebeasi opened the pull request:
<!-- Please refer to our contributing documentation for any questions on submitting a pull request, or let us know here if you need any help: https://ionicframework.com/docs/building/contributing -->

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures


## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying. -->


<!-- Issues are required for both bug fixes and features. -->
Issue URL: Internal ticket

Axe test warns of nested interactive in `ion-segment-button`. This is happening because `ion-segment-button` has `role="tab"` on the host element but then renders a `<button>` in its Shadow DOM. Axe thinks that both the host and the nested button are interactive.

This was added in https://github.com/ionic-team/ionic-framework/pull/23145 before Axe warned about nested interactive (so this was never caught). The original fix was architected this way due to a bug in Chrome on macOS where it does not read the total tab count correctly due to the `role="tab"` elements being in the Shadow DOM.
However, the workaround that avoided this issue introduced the nested interactive issue.

I filed https://bugs.chromium.org/p/chromium/issues/detail?id=1405462 with the Chrome team. I think it is better to avoid the nested interactive issue at the expense of surfacing this Chrome bug. The nested interactive issue is an actual bug that we introduced and impacts all platforms that use assistive tech. The incorrect count bug is a Chrome bug and is limited to Chrome on macOS. The surface area of the Chrome bug is smaller than the nested interactive bug.


## What is the new behavior?
<!-- Please describe the behavior or changes that are being added by this PR. -->

- aria-selected and role states on `ion-segment-button` have been moved to the native `<button>` element inside of the Shadow DOM.
- Segment now calls the internal `setFocus` method on `ion-segment-button` so that the native button element is focused instead of the host element.
- `ariaLabel` on `ion-segment-button` is now inherited to the native `<button>` element.

Notable Changes:

1. Axe will no longer warn about "nested interactives"
2. VoiceOver with Chrome on macOS will now read "tab `n` of 1"  even if there is more than 1 tab in the tab group.

Not impacted by this change:

- TalkBack on Android
- VoiceOver on iOS
- VoiceOver with Safari on macOS
- NVDA on Windows

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

sean-perkins reviewed (approved):
Regressing Chrome macOS in favor of the larger accessibility improvement for other targets, makes complete sense to me.

liamdebeasi commented:
Thanks! Also I added https://github.com/ionic-team/ionic-framework/pull/26575/commits/bb57d37e8830eae81eea99f59317c86864c51c5f which inherits `ariaLabel` to the native button inside of `ion-segment-button` and added a test to a11y (missed this during my initial dev)
